src/ucp/stream/stream_recv.c

```diff
@@ -267,14 +319,18 @@ ucp_stream_am_rdesc_get(ucp_worker_t *worker, void *data, size_t length,
         rdesc = (ucp_recv_desc_t*)ucs_mpool_get_inline(&worker->am_mp);
         if (ucs_likely(rdesc != NULL)) {
             rdesc->flags = 0;
-            memcpy(rdesc + 1, data, length);
+            memcpy(((ucp_stream_am_data_t *)(rdesc + 1))->payload,
```

Comment:
can use ucp_stream_rdesc_am_data() here (need to init rdesc->stream_offset before)

Reply:
then it will add code duplication